crates/pallet-erc20-support/src/migrations/v1.rs

```diff
@@ -0,0 +1,53 @@
+//! Migration to Version 1.
+
+use frame_support::{dispatch::GetStorageVersion, sp_tracing::info, traits::Get, weights::Weight};
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+use crate::BalanceOf;
+use crate::{Approvals, Config, Pallet};
+
+/// Migrate from version 0 to 1.
+pub fn migrate<T: Config<I>, I: 'static>() -> Weight {
+    let current = <Pallet<T, I>>::current_storage_version();
+    let onchain = <Pallet<T, I>>::on_chain_storage_version();
+
+    // Read the onchain version.
+    let mut weight: Weight = T::DbWeight::get().reads(1);
+
+    info!(message = "Running migration to v1", from = ?onchain);
+
+    if onchain == 1 {
+        info!(message = "Already at version 1, nothing to do");
+        return weight;
+    }
+
+    <Approvals<T, I>>::translate(|_owner, _spender, amount: BalanceOf<T, I>| Some(amount.into()));
+
+    // Set new version.
+    current.put::<Pallet<T, I>>();
+
+    // Write the onchain version.
+    weight = weight.saturating_add(T::DbWeight::get().writes(1));
+
+    // Done.
+    weight
+}
+
+/// Check the state before the migration.
+///
+/// Panics if anything goes wrong.
+#[cfg(feature = "try-runtime")]
+pub fn pre_migrate<T: Config<I>, I: 'static>() -> Vec<u8> {
+    sp_std::vec![]
+}
+
+/// Check the state after the migration.
+///
+/// Panics if anything goes wrong.
+#[cfg(feature = "try-runtime")]
+pub fn post_migrate<T: Config<I>, I: 'static>(_state: Vec<u8>) {
+    // Ensure version is updated correctly.
+    let onchain = <Pallet<T, I>>::on_chain_storage_version();
+    assert_eq!(onchain, 1);
```

Comment:
This might not hold if we are migrating on version `2`, in which case the migration will be a noop, but the onchain version will not be `1` (but `2`).

Reply:
We are migration from `0` to `1`.

Reply:
Redesigned based on discussion - https://github.com/humanode-network/humanode/pull/809/commits/e87a191a4722ca14acf015006cb8044b456b50a6

Reply:
https://github.com/humanode-network/humanode/pull/809/commits/d2a676618c492874d6de82720816d12193a4b02a

Reply:
https://github.com/humanode-network/humanode/pull/809/commits/db62ae49b66b28cc6e7afc69d6b9ac0e068c43f5

Reply:
We can remove this fn altogether

Reply:
https://github.com/humanode-network/humanode/pull/809/commits/cd5369271d3d5983bf113f7a46ff47679d54b4d1

Reply:
The issue arises with how one would implement the `post_migrate` calls, given that we how to determine whether to run the migration or not at the call layer. I think in this sense we should only run the `post_migrate` for the *latest* migration, or implement more "atomic" verification calls that could be used in a non-trivial way at the `post_upgrade` level.

Reply:
Thanks, I got your thoughts.